platform/commonUI/edit/src/actions/SaveAsAction.js

```diff
@@ -0,0 +1,169 @@
+            $injector,
+            policyService,
+            dialogService,
+            creationService,
+            copyService,
+            context
+        ) {
+            this.domainObject = (context || {}).domainObject;
+            this.injectObjectService = function(){
+                this.objectService = $injector.get("objectService");
+            };
+            this.policyService = policyService;
+            this.dialogService = dialogService;
+            this.creationService = creationService;
+            this.copyService = copyService;
+        }
+
+        /**
+         * @private
+         */
+        SaveAsAction.prototype.createWizard = function (parent) {
+            return new CreateWizard(
+                this.domainObject,
+                parent,
+                this.policyService
+            );
+        };
+
+        /**
+         * @private
+         */
+        SaveAsAction.prototype.getObjectService = function(){
+            // Lazily acquire object service (avoids cyclical dependency)
+            if (!this.objectService) {
+                this.injectObjectService();
+            }
+            return this.objectService;
+        };
+
+        function resolveWith(object){
+            return function () {
+                return object;
+            };
+        }
+
+        /**
+         * Save changes and conclude editing.
+         *
+         * @returns {Promise} a promise that will be fulfilled when
+         *          cancellation has completed
+         * @memberof platform/commonUI/edit.SaveAction#
+         */
+        SaveAsAction.prototype.perform = function () {
+            // Discard the current root view (which will be the editing
+            // UI, which will have been pushed atop the Browse UI.)
+            function returnToBrowse(object) {
+                if (object) {
+                    object.getCapability("action").perform("navigate");
+                }
+                return object;
+            }
+
+            return this.save().then(returnToBrowse);
+        };
+
+        /**
+         * @private
+         */
+        SaveAsAction.prototype.save = function () {
+            var self = this,
+                domainObject = this.domainObject,
+                copyService = this.copyService;
+
+            function doWizardSave(parent) {
+                var wizard = self.createWizard(parent);
+
+                return self.dialogService
+                    .getUserInput(wizard.getFormStructure(true),
+                        wizard.getInitialFormValue()
+                    ).then(wizard.populateObjectFromInput.bind(wizard));
+            }
+
+            function fetchObject(objectId){
+                return self.getObjectService().getObjects([objectId]).then(function(objects){
+                    return objects[objectId];
+                });
+            }
+
+            function getParent(object){
+                return fetchObject(object.getModel().location);
```

Comment:
Would we be able to instead do this as:

```js
return object.getCapability('location').getOriginal().then(function (original) {
    return original.getCapability('context').getParent();
});
```

This would mean we could reduce dependencies (`$injector` and the `objectService` methods, specifically) and also reduces the amount of specific knowledge here about model properties.

Or would we run into trouble since this is not yet persisted? (I would think not after #773, but not sure)